1.3.c

```diff
@@ -1,121 +1,111 @@
 #include <stdio.h>
+#include <string.h>
 #include <stdbool.h>
+// I am did't do maxsize , because my IDE cant scanf this.
 
-#define MAXIMUMSIZE 256
-
-bool isCorrectString(char* string)
+int IncomingDataStatus(char symbol, char state) 
 {
-    int state = 0;
-    char currentCharacter = -1;
-    int counter = 0;
-    bool isContinue = true;
-    while (isContinue && currentCharacter != '\0')
+    switch (symbol)
     {
-        currentCharacter = string[counter++];
-        switch (state)
+        case 1:
         {
-        case 0:
-            if ((currentCharacter >= 'A' && currentCharacter <= 'Z') || (currentCharacter >= 'a' && currentCharacter <= 'z'))
+            if (symbol >= 'A' && symbol <= 'Z' || symbol >= 'a' && symbol <= 'z')
             {
                 state = 1;
-                break;
             }
-            return false;
-        case 1:
-            if ((currentCharacter >= 'A' && currentCharacter <= 'Z') || (currentCharacter >= 'a' && currentCharacter <= 'z') ||
-                (currentCharacter >= '0' && currentCharacter <= '9') || (currentCharacter == '_'))
+        }
+        case 2:
+        {
+            if (symbol >= '0' && symbol <= '9')
             {
-                break;
+                state = 1;
             }
-            isContinue = false;
-            break;
         }
-    }
-    return currentCharacter == '\0';
-}
+        case 3:
+        {
+            if (symbol == "_")
+            {
+                state = 1;
+            }
 
-bool tests()
-{
-    char testString1[] = "My_var001";
-    char testString2[] = "A";
-    char testString3[] = "_Afewew";
-    char testString4[] = "M_>_";
+        }
+        case 4:
+        {
+            if ((symbol >= 'A' && symbol <= 'Z') || (symbol >= 'a' && symbol <= 'z') || (symbol >= '0' && symbol <= '9') || (symbol == "_"))
+            {
+                state = 1;
+            }
+            else;
+            {
+                state = 2; // state can be 0,1. 2 is false.
 
-    return isCorrectString(testString1) &&
-        isCorrectString(testString2) &&
-        !isCorrectString(testString3) &&
-        !isCorrectString(testString4);
+            }
+        }
+     break;
+    }
+    return state;
 }
 
-#include <stdio.h>
-#include <stdbool.h>
-
-#define maximumSize 256
-
-bool isCorrectString(char* string)
+bool isCorrectString(char *string)
 {
     int state = 0;
-    char currentCharacter = -1;
-    int counter = 0;
-    bool isContinue = true;
-    while (isContinue && currentCharacter != '\0')
+    for (; *string != "/0"; string++)

```

Comment:
```suggestion
    for (; *string != '\0'; string++)
```
Вот так символ конца строки выглядит.
А была строка из двух символов. Фактически, здесь происходило сравнение символа с указателем на строку "/0".
(-1)